index/scorch/segment/zap/segment.go

```diff
@@ -35,7 +35,7 @@ var reflectStaticSizeSegmentBase int
 
 func init() {
 	var sb SegmentBase
-	reflectStaticSizeSegmentBase = int(reflect.TypeOf(sb).Size())
+	reflectStaticSizeSegmentBase = int(unsafe.Sizeof(sb))
```

Comment:
Any idea why this is showing up as a recent failure? This has been in the code base for some time now..

$ go vet $(go list ./... | grep -v vendor/)
index/scorch/segment/zap/segment.go:38: call of unsafe.Sizeof copies lock value: zap.SegmentBase contains sync.Mutex
exit status 1

Reply:
No idea..  reflect.TypeOf(sb).Size() was failing for all the Go versions in CI, With unsafe.Sizeof , it just fails for only one.( the oldest)